ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/ToolsFactory.java

```diff
@@ -0,0 +1,37 @@
+package org.opensearch.ml.engine.tools;
+
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.script.ScriptService;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class ToolsFactory {
+    private static final Map<String, Tool> AGENT_TOOLS = new HashMap<>();
+
+    public ToolsFactory(Client client, ScriptService scriptService, ClusterService clusterService,
+                        NamedXContentRegistry xContentRegistry, Map<String, Tool> externalTools) {
+        AGENT_TOOLS.put(MLModelTool.TYPE, new MLModelTool(client));
+        AGENT_TOOLS.put(MathTool.TYPE, new MathTool(scriptService));
+        AGENT_TOOLS.put(CatIndexTool.TYPE, new CatIndexTool(client, clusterService));
+        AGENT_TOOLS.put(VectorDBTool.TYPE, new VectorDBTool(client, xContentRegistry));
+        AGENT_TOOLS.put(PainlessScriptTool.TYPE, new PainlessScriptTool(client, scriptService));
+        AGENT_TOOLS.putAll(externalTools);
+
+    }
+
+    public Tool getTool(final String toolName) {
+        return  Objects.requireNonNull(AGENT_TOOLS.get(toolName), "Unsupported tool: " + toolName);
+    }
```

Comment:
The keys of AGENT_TOOLS are type, should we change the variable to type?

Reply:
Will rename variable `toolName` -> `agentType`